### DISCUSSION TITLE
Module: Gitlab

### DISCUSSION THREAD

**Discussion by TheTechromancer** [blacklanternsecurity/bbot#1166] [Module Requests]
A module for detecting Gitlab instances, enumerating all their repos, and raising them as `CODE_REPOSITORY` events.

The nice thing about Gitlab instances, since they're self-hosted, is that it's much easier to tell whether they're in-scope.

**Comment by domwhewell-sage** [comment]
Do you know how to use the authenticated API to list repos? Not too familiar with gitlab myself, but like we've done with the github template we might need to make a template class containing both github and gitlab tokens that the git_clone module can inherit from as currently it will only be able to clone authenticated github repos
